V2.0.3

Werri opened the pull request:
fix issues

insom committed:
Add backend connection status flags to stats_mysql_processlist

lorenzocipriani committed:
Fixed path of proxysql.service file

Given that the proxysql.service file has been moved to the subfolder systemd/system, the path in the Makefile had to be updated as well.

carsonip committed:
Fix minor typo

renecannao committed:
Changing version to 2.0.3

renecannao committed:
Merge pull request #1917 from carsonip/fix-typo

Fix minor typo

lorenzocipriani committed:
Added proxysql system user and group management

During the make install phase a proxysql system user and group are created automatically.
As system user this cannot be used for login to the system but just to run the program.

During the make uninstall phase the proxysql user and group are deleted automatically.

Fixed the error with the number of endif statements that prevented the execution of the uninstall rule.

ShiroDN committed:
Fixed systemd service parse error

Fixed systemd service restart specifier parse error. Restart value should be "no", "never" is not acceptable value according to systemd.service man page.

[/lib/systemd/system/proxysql.service:15] Failed to parse service restart specifier, ignoring: never

Commit:
Merge pull request #1925 from ShiroDN/patch-1

Fixed systemd service parse error

Commit:
Merge pull request #1919 from lorenzocipriani/v2.0.2

Fixed path of proxysql.service file

pondix commented:
Automated message: PR pending admin approval for build testing

pondix commented:
hi @Werri - these are fixes from v2.0.3, why are you trying to merge them into v2.0.2?

- v2.0.2 is final and has been released, anything new must go into v2.0.3

renecannao committed:
Do not kill Session if a backend failed

This new algorithm makes the backend connection fails instead of killing the session
This is an improvement over issue #1085

renecannao committed:
Fix bug #1927

renecannao committed:
Merge pull request #1932 from sysown/v2.0.3-1927

Fix bug #1927

renecannao committed:
Merge pull request #1900 from insom/status_flags

Add backend connection status flags to stats_mysql_processlist

renecannao committed:
Minor code cleanup #1882

carsonip committed:
Fix skipped mypoll processing

carsonip committed:
Fix skipping sessions in idle threads iteration

renecannao committed:
Merge pull request #1945 from carsonip/fix-mypoll

Fix skipped mypoll processing

renecannao committed:
Merge pull request #1948 from carsonip/fix-idle-session-loop

Fix skipping sessions in idle threads iteration

carsonip committed:
Do not move throttled session to epoll

Epoll thread does not poll for EPOLLOUT. If a throttled session with
pending data out is moved to epoll thread, the session will stay in
epoll thread indefinitely, causing a hang observed by client. Fix by not
moving throttled session to epoll thread.

Fix #1939

carsonip committed:
No throttle when throttle_max_bytes_per_second_to_client == 0

If throttle_max_bytes_per_second_to_client == 0, disable throttling.
This matches the mysql convention of 0 == no limit. Change the default
to 0.

renecannao committed:
Fix bug in authentication after AuthSwitch

If AuthSwitchResponse has a NULL char, parsing fails.
This is likely the root cause of #1944 , but also #1934

renecannao committed:
Merge pull request #1952 from carsonip/fix-epoll-hang-2

Do not move throttled session to epoll

renecannao committed:
Merge pull request #1953 from carsonip/fix-throttle

No throttle when throttle_max_bytes_per_second_to_client == 0

renecannao committed:
Merge pull request #1954 from sysown/v2.0.3-1944

Fix bug in authentication after AuthSwitch

renecannao commented:
Closing
